core/pkg/artifacts/saver.go

```diff
@@ -720,3 +739,32 @@ func (as *ArtifactSaver) Save() (artifactID string, rerr error) {
 
 	return artifactID, nil
 }
+
+const minArtifactTagsServerVersion = "v0.58"
+
+// canSupportArtifactTags returns true if the server version supports artifact tags.
+func (as *ArtifactSaver) canSupportArtifactTags() (bool, error) {
+	response, err := gql.ServerInfo(as.Ctx, as.GraphqlClient)
+	if err != nil {
+		return false, fmt.Errorf("gql.ServerInfo: %w", err)
+	}
+
+	// If isn't a versioned server deployment (as opposed to e.g. prod SaaS), err on the side of leniency, and
+	// expect artifact tags support in more recent deployments
+	serverInfo := response.GetServerInfo()
+	if serverInfo == nil {
+		return true, nil
```

Comment:
i know that you are just trying to make it work for artifact tags, but this is not great, since other queries might also need this treatment, ideally if you could extract this logic to generalize it for other queries as well, would've been a better way to go about it here

Reply:
Actually just realized -- with the introspection query happening earlier in Python, this is no longer even needed for tags.

I can try to generalize this and put it somewhere more appropriate (let me know if there's a preference/recommendation for where)

Reply:
that's exactly the issue, there is a sketch but your code is actually the first one that needs it so part of the task is to design it work in general.

Reply:
> there is a sketch but your code is actually the first one that needs it so part of the task

Just clarifying when you mention there's a sketch -- is this in the code elsewhere or is there another doc/plan/ticket already related to this?  (and or proposing it)

Reply:
no doc, just what is described in the contributing doc... where we create multiple version of the same query per version, but it seems maybe you are using the auto generated directive to come up with something better.

Now the thing that is still an issue, is that you only solved it for artifact, but it is not re-usable for any other dev:
- querying for version, will be done every time, since we don't save the version value. 
- given that the version is shared, need a way to have some logic to handle different versions

Reply:
Genuinely wondering -- does it make sense to apply `omitempty: true` by default?  Not a perfect solution, just a partial one -- doesn't prevent compatibility issues from all schema changes, just those like this that add new fields

Reply:
Separately -- don't get me wrong, I totally understand the utility of renaming + parameterizing + making this check on server info more general.

That said, if it turns out it's not a really quick change I might prefer to separate it from this PR -- I'm concerned it won't be useful in practice, since other devs will need other workaround (for now) won't help pass that one CI check against the min server version

Reply:
following offline discussion earlier this morning -- it sounds like `versionOnThisInstanceString` may not be the way to check for server compatibility in the future.  considering this function is also currently redundant, I think I'd rather remove this helper instead of generalizing it, just to keep things simple for now